### PR TITLE
Properly ignore files during deserialize

### LINKED ISSUE #5426: Deserialize does not ignore files properly
If a repo contains a file more than one and then the deserialize process might associate the wrong file with the object. 

Example:

```abap
/src/
  /src/zcl_clock.clas.abap
  /src/.meta/zcl_clock.clas.abap 

ignore `*/.meta/*`
```

The status calculation and the deserialize process do not ignore the file in `.meta`. Therefore, the class will be created based on the wrong file.

### PR DESCRIPTION
**Pull request by mbtools**
Closes #5426

This moves the ignore logic from `zcl_abapgit_files_status->prepare_remote` to an option of `zcl_abapgit_repo->get_files_remote`. Deserialize can then ignore the files already when getting them via `get_files_remote`.

PS: Required for https://github.com/exercism/abap/pull/82

### CODE REVIEW

**Comment by larshp** [comment]
@mbtools check conflicts

File: src/repo/zif_abapgit_repo.intf.abap [hunk 1]
```diff
@@ -31,16 +31,18 @@ INTERFACE zif_abapgit_repo
       zcx_abapgit_exception .
   METHODS get_files_remote
     IMPORTING
-      ii_obj_filter   TYPE REF TO zif_abapgit_object_filter OPTIONAL
+      !ii_obj_filter   TYPE REF TO zif_abapgit_object_filter OPTIONAL
+      !iv_ignore_files TYPE abap_bool DEFAULT abap_false
+        PREFERRED PARAMETER ii_obj_filter
```
Comment (larshp) [inline src/repo/zif_abapgit_repo.intf.abap:36]: wondering, what is the scenario where we dont want to remove the ignored files? can this parameter be removed? should it be default true?

Comment (mbtools) [reply src/repo/zif_abapgit_repo.intf.abap:36]: Long term, I can image showing all remote files in AG (like VScode). You could view or even update all of them in AG, too (like readme etc). Then have a view option to hide "ignored files".

**Comment by cshbicos** [comment]
Is it possible this merge killed the configuration file for the [XML Stripper plugin](https://github.com/sbcgua/abapgit_xml_stripper_plugin) that is mentioned in the [User Exit](https://docs.abapgit.org/ref-exits.html#pre_calculate_repo_status) page on the website?

It appears if I want to have the /.xmlstrip.config file in my repository, by the time PRE_CALCULATE_REPO_STATUS is run, the file has already been filtered out. From the change log, it gather that before this change, I would have a had a chance to get to all the files in PRE_CALCULATE_REPO_STATUS

Is there an alternative way to maintain additional files like that moving forward? I would like to avoid hard coding these type of settings in my user exit.

**Comment by mbtools** [comment]
You are right. The xml stripper does not work anymore using the file in root (it shouldn't have relied on files that are ignored by abapGit). 

I suggest moving the config file to the /arc folder and changing the xml stripper accordingly:

https://github.com/sbcgua/abapgit_xml_stripper_plugin/blob/master/src/zcl_abapgit_xml_stripper.clas.abap change '/' to '/src' in the find method.

**Pull request merged**